subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -96,6 +97,41 @@ static void scan_result_cb(struct net_if *iface, int status,
 #endif /* CONFIG_WIFI_MGMT_RAW_SCAN_RESULTS_ONLY */
 }
 
+static int parse_scan_bands(char *scan_bands_str,
+							uint8_t *band_map)
+{
+	char *band = NULL;
+	uint8_t parsed_band_map = 0;
+
+	if (!scan_bands_str) {
+		return -EINVAL;
+	}
+
+	band = strtok(scan_bands_str, ",");
```

Comment:
Hmm, I am not sure about using strtok here, I recall its use should be avoided because its error prone. Do we need to separate elements with , in the first place?

Reply:
> Do we need to separate elements with , in the first place?

Considering the multitude of new options added (in subsequent commits), delimiter based parsing is working out to be better and more user friendly than positional parsing.